zapetw/write_syncer_windows.go

```diff
@@ -0,0 +1,67 @@
+package zapetw
+
+import (
+	"github.com/Microsoft/go-winio/pkg/etw"
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+)
+
+const providername = "ACN-Data-Plane"
+
+type ETWWriteSyncer struct {
+	provider  *etw.Provider
+	eventName string
+	etwLevel  etw.Level
+}
+
+func NewETWWriteSyncer(eventName string, zapLevel zapcore.Level) (*ETWWriteSyncer, error) {
+	provider, err := etw.NewProviderWithOptions(providername)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create ETW provider")
+	}
+
+	return &ETWWriteSyncer{
+		provider:  provider,
+		eventName: eventName,
+		etwLevel:  mapZapLevelToETWLevel(zapLevel),
+	}, nil
+}
+
+func (e *ETWWriteSyncer) Write(p []byte) (int, error) {
+	err := e.provider.WriteEvent(
+		e.eventName,
+		etw.WithEventOpts(
+			etw.WithLevel(e.etwLevel),
+		),
+		[]etw.FieldOpt{
+			etw.StringField("Message", string(p)),
+		},
+	)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to write to ETW")
```

Comment:
should be errors.Wrap(err, "failed to write to ETW") since we do not need pass args here